### PR TITLE
Changed targeting from .NET 4.5.1 to .NET 4.5.2

### PR DESCRIPTION
**Pull request by michaelnoonan**
I started down this path because I couldn't build.
All the same OS support all patches of .NET 4.5 including .NET 4.5.2.
I cleaned up all the build/pack warnings.
Relates to https://github.com/OctopusDeploy/OctopusClients/pull/523

### CODE REVIEW

File: BuildAssets/OctopusTools.nuspec [hunk 1]
```diff
@@ -15,9 +15,10 @@
     <releaseNotes></releaseNotes>
     <language>en-US</language>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
-    <licenseUrl>https://github.com/OctopusDeploy/OctopusCLI/blob/master/LICENSE.txt</licenseUrl>
+    <license type="expression">Apache-2.0</license>
```
Comment (michaelnoonan) [inline BuildAssets/OctopusTools.nuspec:18]: `licenseUrl` is deprecated: https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5125

File: BuildAssets/OctopusTools.nuspec [hunk 2]
```diff
@@ -15,9 +15,10 @@
...
     <projectUrl>https://github.com/OctopusDeploy/OctopusCLI/</projectUrl>
-    <iconUrl>http://i.octopusdeploy.com/resources/Avatar3_360.png</iconUrl>
+    <repository type="git" url="https://github.com/OctopusDeploy/OctopusClients" />
+    <icon>tools\icon.png</icon>
```
Comment (michaelnoonan) [inline BuildAssets/OctopusTools.nuspec:21]: `iconUrl` is deprecated: https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5048

File: source/Octo/Octo.csproj [hunk 3]
```diff
@@ -17,11 +17,15 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">
```
Comment (michaelnoonan) [inline source/Octo/Octo.csproj:20]: The runtime identifiers were missing for `netcoreapp2.0`

File: source/Octopus.Cli/Octopus.Cli.csproj [hunk 4]
```diff
@@ -33,7 +27,7 @@
     <PackageReference Include="NuGet.Packaging.Core" Version="3.6.0-octopus-58692" />
     <PackageReference Include="NuGet.Packaging.Core.Types" Version="3.6.0-octopus-58692" />
     <PackageReference Include="NuGet.Versioning" Version="3.6.0-octopus-58692" />
-    <PackageReference Include="Octopus.Client" Version="8.2.0" />
+    <PackageReference Include="Octopus.Client" Version="8.4.1-michaelnoonan-targ-5" />
```
Comment (michaelnoonan) [inline source/Octopus.Cli/Octopus.Cli.csproj:30]: Temporary reference to prove it works.

File: source/Octo/Octo.csproj (hunk 3 above)
Comment (droyad) [reply source/Octo/Octo.csproj:20]: What are we even building `netcoreapp2.x`?

Comment (droyad) [reply source/Octo/Octo.csproj:20]: Ah I see, next file. The runtimes should only be needed for the project that is the executable

Comment (michaelnoonan) [reply source/Octo/Octo.csproj:20]: Looks like we are according to the build file. Sounds like we will phase it out over time: https://github.com/OctopusDeploy/OctopusCLI/blob/master/build.cake#L144

Comment (droyad) [reply source/Octo/Octo.csproj:20]: Ah, the `.portable` build. We shouldn't need the runtimes then.

**Pull request merged**